### ISSUE TITLE
[Bug] GTK errors when running + no window opens

### ISSUE THREAD

**Issue by solonovamax** [Suwayomi/Suwayomi-Server#38]
I acknowledge that:

- [x] I have updated to the latest version of the app.
- [x] I have tried the troubleshooting guide described in `README.md`
- [x] If this is a request for adding/changing an extension it should be brought up to Tachiyomi: https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose
- [x] If this is an issue with some extension not working properly, It does work inside Tachiyomi as intended.
- [x] I have searched the existing issues and this is a new ticket **NOT** a duplicate or related to another open issue
- [x] I will fill out the title and the information in this template

---

Yarn should be listed as a dependency in the README, as well as having good error logging.

## Device information
- Tachidesk version: `v0.2.5-r256-jar` (edit: just confirmed this still occurs on `v0.2.6-r273-jar`).
- Server Operating System: Arch Linux 5.11.8-arch1-1
- Server JVM version: openjdk 11.0.10 2021-01-19, OpenJDK Runtime Environment (build 11.0.10+9)
- Client Operating System: Arch Linux 5.11.8-arch1-1
- Client Web Browser: Chromium 89.0.4389.90 Arch Linux (Using the Ungoogled Chromium fork of Chromium)

## Steps to reproduce
1. Download & Run jar
2. GTK Errors & other stuff is logged. No windows open.
3. <kbd>Control</kbd> + <kbd>C</kbd> to exit.

### Expected behavior
The application actually works

### Actual behavior
It doesn't.

## Other details
Here's the log:
```
$ java -jar Tachidesk-v0.2.5-r256.jar 
[main] INFO Exposed - Preparing create tables statements took 30ms
[main] INFO Exposed - Executing create tables statements took 3ms
[main] INFO Exposed - Extracting table columns took 73ms
[main] INFO Exposed - Extracting column constraints took 15ms
[main] INFO Exposed - Preparing alter table statements took 89ms
[main] INFO Exposed - Executing alter table statements took 0ms
[main] WARN Exposed - Indices exist in database and not mapped in code on class 'CHAPTER':
[main] WARN Exposed - 		Index 'FK_CHAPTER_MANGA_ID_INDEX_5' for 'CHAPTER' on columns ir.armor.tachidesk.database.table.ChapterTable.manga
[main] WARN Exposed - Indices exist in database and not mapped in code on class 'SOURCE':
[main] WARN Exposed - 		Index 'FK_SOURCE_EXTENSION_ID_INDEX_9' for 'SOURCE' on columns ir.armor.tachidesk.database.table.SourceTable.extension
[main] WARN Exposed - Indices exist in database and not mapped in code on class 'PAGE':
[main] WARN Exposed - 		Index 'FK_PAGE_CHAPTER_ID_INDEX_2' for 'PAGE' on columns ir.armor.tachidesk.database.table.PageTable.chapter
[main] WARN Exposed - Indices exist in database and not mapped in code on class 'CATEGORYMANGA':
[main] WARN Exposed - 		Index 'FK_CATEGORYMANGA_CATEGORY_ID_INDEX_C' for 'CATEGORYMANGA' on columns ir.armor.tachidesk.database.table.CategoryMangaTable.category
[main] WARN Exposed - 		Index 'FK_CATEGORYMANGA_MANGA_ID_INDEX_C' for 'CATEGORYMANGA' on columns ir.armor.tachidesk.database.table.CategoryMangaTable.manga
[main] INFO Exposed - Checking mapping consistence took 14ms
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by dorkbox.util.Swt (file:/home/solonovamax/Downloads/Tachidesk-v0.2.5-r256.jar) to method java.lang.ClassLoader.findLoadedClass(java.lang.String)
WARNING: Please consider reporting this to the maintainers of dorkbox.util.Swt
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[main] WARN dorkbox.util.jna.linux.GtkTheme - Unable to get tray image size. Using default.

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.182: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): Gdk-CRITICAL **: 12:13:43.189: gdk_x11_get_xatom_by_name_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(java:343959): dbind-WARNING **: 12:13:43.293: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-6kGghwTyak: No such file or directory
^C^C^C
```

**Comment by solonovamax** [comment]
wait, forgot to add title lol

**Comment by AriaMoradi** [comment]
Try the jar built with cfc6e5c, someone had a similar issue which that commit fixed

**Comment by solonovamax** [comment]
I already built from source and tried with that fix, and it did not work (I edited my error earlier to say that I tried it with the latest jar build from source, which was `v0.2.6-r273-jar`.)

But I'll build it again and try once more, because why not.

**Comment by solonovamax** [comment]
Still has same issues. If I try to connect to `localhost:4567`, there's nothing.

**Comment by AriaMoradi** [comment]
After some pair examination with @solonovamax It was concluded that the issue arises from Dorkbox SystemTray, the System Tray library this project uses and it has to be fixed upstream.
Temperary fix: 
Method 1: Change your gtk theme to default (in case of @solonovamax, they were using the gtk Breeze theme) 
Method 2: disable the feature by commenting out this block of code inside `ServerSetup.kt`
```kotlin
    try {
        systemTray
    } catch (e: Exception) {
        e.printStackTrace()
    }
```

**Comment by solonovamax** [comment]
you should probably tag this with something along the lines of "waiting on 3rd party"

**Comment by AriaMoradi** [comment]
SystemTray dependency was recently updated, is the issue fixed?
Try with v0.3.0

**Comment by AriaMoradi** [comment]
The source of this issue is the SystemTray library, nothing that tachidesk can fix.
